storage/Devices/StrayBlkDevice.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2018 SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact Novell, Inc.
+ *
+ * To contact Novell about this file by physical or electronic mail, you may
+ * find current contact information at www.novell.com.
+ */
+
+
+#ifndef STORAGE_STRAY_BLK_DEVICE_H
+#define STORAGE_STRAY_BLK_DEVICE_H
+
+
+#include "storage/Devices/BlkDevice.h"
+
+
+namespace storage
+{
+
+    /**
+     * This class represents stray block devices. Stray block devices
+     * have no parent in the devicegraph and cannot be partitioned.
+     *
+     * Normally only Disks and Dasds have no parents in the
+     * devicegraph but Disks and Dasds can be partitioned.
+     *
+     * Currently only Xen Virtual Partitions are represented as stray
+     * block devices (e.g. /dev/xvda1). Note that partitionable
+     * devices in Xen are represented as Disks (e.g. /dev/xvda) and
+     * partitions on them as normal Partitions (e.g. also /dev/xvda1).
```

Comment:
So, while we are adding this for the case when xvda1 has no parent, OTOH in some deployments, xvda1 is a partition on a xvda disk, just like with sda and vda. Right?